.jenkins/Jenkinsfile

```diff
@@ -12,6 +12,8 @@ REPOSITORY_NAME = env.REPOSITORY_NAME ?: "openenclave/openenclave"
 BRANCH_NAME = env.BRANCH_NAME ?: "master"
 DOCKER_TAG = env.DOCKER_TAG ?: "latest"
 FULL_TEST_SUITE = env.FULL_TEST_SUITE ?: false
+// Regex that includes directory you want to ignore for CI builds.
+String IGNORED_DIRS = "docs|\\.jenkins/infrastructure"
```

Comment:
Do we ignore the infrastructure directory because it only contains jobs not run in the BORS jobs? If so, is there a way we can do some validation of the infrastructure code to make sure the pipelines work?

Reply:
I'm glad you asked as this ties into some plans for the future.

Right now my reasoning to ignore .jenkins/infrastructure is that it contains infrastructure Dockerfiles, Jenkinsfiles, and other provisioning scripts that are not ran by our usual Bors CI tests. There are also no readily available tests we have for infrastructure code yet. So I believe there is no point in running Bors CI tests for .jenkins/infrastructure changes at the moment.

In the future, when infrastructure tests become available, it is then possible to use the same setup as this PR to not only ignore directories, but to also conduct changes in certain directories to run certain sets of tests. E.g. changes in .jenkins/infrastructure/docker will run docker builds.